web/html/src/manager/notifications/notifications-list.tsx

```diff
@@ -0,0 +1,454 @@
+import * as React from "react";
+
+import { AsyncButton, LinkButton } from "components/buttons";
+import { Dialog } from "components/dialog/Dialog";
+import { Form, Select } from "components/input";
+import { Messages as MessageContainer, MessageType, Utils as MessagesUtils } from "components/messages/messages";
+import { TopPanel } from "components/panels/TopPanel";
+import { SectionToolbar } from "components/section-toolbar/section-toolbar";
+import { Column } from "components/table/Column";
+import { SearchField } from "components/table/SearchField";
+import { Table } from "components/table/Table";
+
+import { stringToReact } from "utils";
+import { Utils } from "utils/functions";
+import Network from "utils/network";
+
+export type NotificationType = {
+  label: string;
+  description: string;
+};
+
+enum Severity {
+  Info = "info",
+  Warning = "warning",
+  Error = "error",
+}
+
+type Notification = {
+  id: number;
+  severity: Severity;
+  type: string;
+  summary: string;
+  details?: string;
+  read: boolean;
+  actionable: boolean;
+  create: Date;
+};
+
+type Props = {
+  notificationTypes: Array<NotificationType>;
+};
+
+type State = {
+  serverData: Array<Notification> | undefined;
+  dataUrlTags: Array<string>;
+  currentDataUrlTag: string;
+  loading: boolean;
+  messages: Array<MessageType>;
+  selectedItems: Array<number>;
+  popupItem: Notification | undefined;
+  typeCriteria: Array<string>;
+};
+
+export class NotificationList extends React.Component<Props, State> {
+  private readonly typeMap: Map<string, string>;
+
+  private readonly sortedNotificationTypes: Array<NotificationType>;
+
+  public constructor(props: Props) {
+    super(props);
+
+    // Sort the notification types by description
+    this.sortedNotificationTypes = [...this.props.notificationTypes].sort((a, b) =>
+      a.description.localeCompare(b.description)
+    );
+
+    // Map to convert from label to description
+    this.typeMap = this.props.notificationTypes.reduce(
+      (map, item) => map.set(item.label, item.description),
+      new Map<string, string>()
+    );
+
+    this.state = {
+      serverData: undefined,
+      dataUrlTags: ["#data-unread", "#data-all"],
+      currentDataUrlTag: window.location.hash ? window.location.hash : "#data-unread",
+      loading: true,
+      messages: [],
+      selectedItems: [],
+      popupItem: undefined,
+      typeCriteria: [],
+    };
+  }
+
+  public componentDidMount(): void {
+    this.refreshServerData(this.state.currentDataUrlTag);
+  }
+
+  public componentDidUpdate(prevProps: Readonly<Props>, prevState: Readonly<State>): void {
```

Comment:
I see this and the related code around `dataUrlTags` was already there in the old code that was there previously. I'm pretty certain we don't want this approach, but I'll have to get back to you once I learn how our current spa engine implementation works. In the meanwhile we're fine to proceed as-is, but just so you know, we don't want to replicate this elsewhere or to new code etc.

Reply:
Honestly, I just left this code as is because I was not sure to fully understand what it was doing. Now I ha a second look and I saw that if we manually tamper with url the existing logic breaks. So I now simplified how we handle the hash and the selected tab. Not sure if your comment was only referring to that or if it was also about the `componentDidMount()`/`componentDidUpdate()` strategy. If that's the case, let me know how we should handle it in the correct way :-)

Reply:
My comment was about the hash situation, but I've created a separate ticket to look into that in a wider sense, so feel free to disregard that topic under this PR.